src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -84,6 +84,20 @@ public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsAsync(Diagnosti
                 return SpecializedCollections.EmptyEnumerable<DiagnosticData>();
             }
 
+            if (analyzer == GeneratorDiagnosticsPlaceholderAnalyzer.Instance)
+            {
+                // We will count generator diagnostics as semantic diagnostics; some filtering to either syntax/semantic is necessary or else we'll report diagnostics twice.
+                if (kind == AnalysisKind.Semantic)
+                {
+                    var generatorDiagnostics = await textDocument.Project.GetSourceGeneratorDiagnosticsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
So this method executes in-proc, do we want to run `GetSourceGeneratorDiagnosticsAsync` also in-proc?

Reply:
We're practically running generators in-proc too (since we need compilations there), but happy to move the work OOP for consistency here. I guess I must have missed then for the document analysis case where we were jumping OOP then? Where is that?

Reply:
> for the document analysis case where we were jumping OOP then

We run the internal DocumentDiagnosticAnalyzers and ProjectDiagnosticAnalyzers in-proc, as this would need the partner teams that have authored these analyzers to also support running out of proc. We do run all the first party and third party analyzers out-of-proc: https://github.com/dotnet/roslyn/blob/acb47af8bc29d116d73d1e3469d9a5a625e9fcae/src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs#L151-L166

Reply:
Chatting with @mavasani we realized cleaning this up properly is going to require a bit more work here: when we're doing FSA analysis OOP, we are creating in-proc Compilations even if we don't end up using them. Before the assumption was that relatively cheap (since creating a Compilation didn't really do anything if you didn't ask it for anything), but now with source generators producing a Compilation actually is going to do work. Since we don't have many users using FSA, and that problem isn't new with this PR, we're going to count fixing that as out of scope here, but will follow up next week.